src/docbuilder/rustwide_builder.rs

```diff
@@ -322,14 +330,18 @@ impl RustwideBuilder {
 
         let local_storage = tempfile::Builder::new().prefix("docsrs-docs").tempdir()?;
 
+        let metadata = Metadata::from_crate_root(&build_dir.build_dir())?;
```

Comment:
Hmm, are you sure this is right? I think builds happen in a temporary directory, not in the source directory.

Reply:
I haven't tested it, but `Build::host_source_dir` returns `self.dir.source_dir()` so i just made that method public.

Reply:
You have build_dir here, not source_dir - is that intentional?

Reply:
Unfortunately this won't work. `source_dir()` is populated during the `run()` call ([`build.rs`](https://github.com/rust-lang/rustwide/blob/7a4ef73f58378f7219f996ca8c6d9497d178f7b6/src/build.rs#L151-L152) => [`prepare.rs`](https://github.com/rust-lang/rustwide/blob/7a4ef73f58378f7219f996ca8c6d9497d178f7b6/src/prepare.rs#L39)), so it's going to be empty where you call it. What we could do instead is to make rustwide's [`Crate::copy_source_to`](https://github.com/rust-lang/rustwide/blob/7a4ef73f58378f7219f996ca8c6d9497d178f7b6/src/crates/mod.rs#L76-L85) method public, copy the source to a temporary directory, fetch the metadata from it, and then delete the temporary directory.

Reply:
how about a BuildDir::create_or_get_source_dir (or whatever name) which both docs.rs and run() can call so we don't waste cpu cycles moving everything twice